chips/nrf52/src/ieee802154_radio.rs

```diff
@@ -767,128 +797,237 @@ impl<'a> Radio<'a> {
+            // It should not be possible to receive an interrupt while the tracked radio state is OFF
+            RadioState::OFF => panic!("Received interrupt while off"),
+            RadioState::RX => {
+                // Enforce READY_START shortcut enablement if radio is receiving
+                if !self.registers.shorts.is_set(Shortcut::READY_START) {
+                    panic!("READY_START shortcut not enabled while sending.")
+                }
 
-        if self.registers.event_ccabusy.is_set(Event::READY) {
-            self.registers.event_ccabusy.write(Event::READY::CLEAR);
-            self.registers.event_ready.write(Event::READY::CLEAR);
-            self.registers.task_disable.write(Task::ENABLE::SET);
-            while self.registers.event_disabled.get() == 0 {}
-            self.registers.event_disabled.write(Event::READY::CLEAR);
-            //need to back off for a period of time outlined
-            //in the IEEE 802.15.4 standard (see Figure 69 in
-            //section 7.5.1.4 The CSMA-CA algorithm of the
-            //standard).
-            if self.cca_count.get() < IEEE802154_MAX_POLLING_ATTEMPTS {
-                self.cca_count.set(self.cca_count.get() + 1);
-                self.cca_be.set(self.cca_be.get() + 1);
-                let backoff_periods = self.random_nonce() & ((1 << self.cca_be.get()) - 1);
-                self.timer0
-                    .unwrap_or_panic() // Unwrap fail = Missing timer reference for CSMA
-                    .set_alarm(
-                        kernel::hil::time::Ticks32::from(0),
-                        kernel::hil::time::Ticks32::from(
-                            backoff_periods * (IEEE802154_BACKOFF_PERIOD as u32),
-                        ),
-                    );
-            } else {
-                self.transmitting.set(false);
-                //if we are transmitting, the CRCstatus check is always going to be an error
-                let result = Err(ErrorCode::BUSY);
-                //TODO: Acked is flagged as false until I get around to fixing it.
-                self.tx_client.map(|client| {
-                    let tbuf = self.tx_buf.take().unwrap(); // Unwrap fail = TX Buffer produced error when sending it back to the requestor after the channel was busy.
-                    client.send_done(tbuf, false, result)
-                });
+                // Since READY_START shortcut enabled, always clear READY event
+                self.registers.event_ready.write(Event::READY::CLEAR);
+
+                // Completed receiving a packet, now determine if we need to send ACK
+                if self.registers.event_end.is_set(Event::READY) {
+                    self.registers.event_end.write(Event::READY::CLEAR);
+                    let result = self.crc_check();
+
+                    // Unwrap fail = RX Buffer produced error when sending received packet to requestor
+                    let rbuf = self.rx_buf.take().unwrap();
```

Comment:
Looking further into this, I do not think this has to do with the DMA event. The only instance this would occur is if the client receive(...) method did not call `self.mac.set_receive_buffer(buf)` to return the buffer to the radio driver. I may be overlooking something related to the DMA event, but this is my understanding.

The set_dma_ptr method always returns the buffer, so the only case the buffer "wouldn't be returned from DMA" is if someone forgets to replace the buffer after the set_dma_ptr method is called. Please let me know if the updated comments are more clear.